snapcraft/remote/launchpad.py

```diff
@@ -160,28 +160,30 @@ def _fetch_artifacts(self, snap: Entry) -> None:
             self._download_build_artifacts(build)
             self._download_log(build)
 
-    def _get_builds_collection_entry(self, snap: Entry) -> Optional[Entry]:
+    def _get_builds_collection_entry(self, snap: Optional[Entry]) -> Optional[Entry]:
         logger.debug("Fetching builds collection information from Launchpad...")
-        url = snap.builds_collection_link
-        return self._lp_load_url(url)
+        url = snap.builds_collection_link  # type: ignore
+        return self._lp_load_url(cast(str, url))
```

Comment:
```suggestion
        url = cast(str, snap.builds_collection_link)
        return self._lp_load_url(url)
```

Reply:
Oo thanks.  You got me on the right track, check it out now.

Reply:
Fantastic!